src/compile/data/transforms.ts

```diff
@@ -18,18 +18,20 @@ import {TimeUnitNode} from './timeunit';
 
 
 export class FilterNode extends DataFlowNode {
+  private expr: string;
   public clone() {
     return new FilterNode(this.model, duplicate(this.filter));
   }
 
   constructor(private readonly model: Model, private filter: LogicalOperand<Filter>) {
     super();
+    this.expr = expression(this.model, this.filter, this);
```

Comment:
For some reason, FilterNode already previously stored model even though it wasn't previously used.  (Note that this file is now in `filter.ts` as of July 26, 2018)